pkg/client/rp/relying_party_test.go

```diff
@@ -105,3 +111,82 @@ func Test_verifyTokenResponse(t *testing.T) {
 		})
 	}
 }
+
+func Test_PKCEFromDiscovery(t *testing.T) {
+	tests := []struct {
+		name     string
+		methods  []string
+		expected bool
+	}{
+		{name: "nil", methods: nil, expected: false},
+		{name: "empty", methods: []string{}, expected: false},
+		{name: "invalid", methods: []string{"invalid"}, expected: false},
+		{name: "plain", methods: []string{"plain"}, expected: true},
+		{name: "S256", methods: []string{"S256"}, expected: true},
+		{name: "both", methods: []string{"plain", "S256"}, expected: true},
+		{name: "mixed", methods: []string{"invalid", "S256"}, expected: true},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			server := httptest.NewServer(
+				http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+					if r.URL.Path != oidc.DiscoveryEndpoint {
+						w.WriteHeader(http.StatusNotFound)
+						return
+					}
+
+					if err := json.NewEncoder(w).Encode(map[string]interface{}{
+						"issuer":                           "http://" + r.Host,
+						"code_challenge_methods_supported": tt.methods,
+					}); err != nil {
+						t.Fatalf("unexpected error encoding '%v' to JSON: %v", tt.methods, err)
+					}
+				}))
+			defer server.Close()
+
+			t.Log("issuer", server.URL)
+
+			clientID := t.Name() + "-client"
+			clientSecret := t.Name() + "-secret"
+			targetURL := "http://local-site"
+			rp, err := NewRelyingPartyOIDC(
+				t.Context(),
```

Comment:
We currently still support go 1.23. So test context can't be used yet.


```suggestion
				context.Background(),
```